apis/network/v1beta1/dnsdata_types.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DNSHost holds the mapping between IP and hostnames that will be added to dnsmasq hosts file.
+type DNSHost struct {
+	// +kubebuilder:validation:Required
+	// IP address of the host file entry.
+	IP string `json:"ip"`
+
+	// +kubebuilder:validation:Required
+	// Hostnames for the IP address.
+	Hostnames []string `json:"hostnames"`
+}
+
+// DNSDataSpec defines the desired state of DNSData
+type DNSDataSpec struct {
+	// +kubebuilder:validation:Optional
+	Hosts []DNSHost `json:"hosts,omitempty"`
```

Comment:
It's weird that having both `Optional` and `omitempty` doesn't fail `operator-lint` in `pre-commit` in Prow.  I thought we checked for that.  Maybe this repo is using an older `operator-lint` version.

Reply:
isn't that only if there is a default value specified? https://github.com/gibizer/operator-lint/tree/main/linters/crd/C003

Reply:
Ah, I did not know that.  Fair enough.